liquidity_pool/src/lib.rs

```diff
@@ -218,8 +233,38 @@ impl LiquidityPoolTrait for LiquidityPool {
         put_reserve_b(&e, balance_b);
     }
 
-    fn swap(e: Env, to: Identifier, out_a: i128, out_b: i128) {
+    fn swap(e: Env, to: Address, buy_a: bool, out: i128, in_max: i128) {
+        to.require_auth();
+
         let (reserve_a, reserve_b) = (get_reserve_a(&e), get_reserve_b(&e));
+
+        let reserve_sell: i128;
```

Comment:
nit: you can do `let (reserve_sell, reserve_buy) = if buy_a { (reserve_b, reserve_a) } else { (reserve_a, reserve_b) };` (more idiomatic IMHO)

Reply:
done